actions/views/root.js

```diff
@@ -8,18 +8,21 @@ import {Client4} from 'mattermost-redux/client';
 import {ActionTypes} from 'utils/constants';
 
 export function loadMeAndConfig() {
-    return (dispatch) => {
-        // if any new promise needs to be added please be mindful of the order as it is used in root.jsx for redirection
+    return async (dispatch) => {
+        // need to await for clientConfig first as it is required for loadMe
+        const clientConfig = await dispatch(getClientConfig());
```

Comment:
@chetanyakan, is it safe to dispatch `getLicenseConfig` asynchronously?

Reply:
@lieut-data I am not sure. Could you elaborate on why it would not be safe?

Reply:
This will slow down the initiation of the app as we are waiting for `getClientConfig` response. Does it not work if we pass the key value from config to the react component it requires for conditional updates?

Reply:
One alternative is to get [terms of service](https://github.com/mattermost/mattermost-redux/blob/master/src/actions/users.js#L244) and [custom emoji](https://github.com/mattermost/mattermost-redux/blob/master/src/actions/users.js#L250) always regardless of client config value in mattermost-redux.

Reply:
@lieut-data @sudheerDev Would it make more sense to make this change instead? 

Also, I am not sure about the working of custom emoji. So, could you help me understand if changing that would be required.

Reply:
@chetanyakan Could you help me out in understanding how this fixes the issue in the ticket?

Reply:
Aah i think i get it. The value of selector is defaulted to false so if config does not exist then you would be seeing this state?

Reply:
@sudheerDev [This API call](https://github.com/mattermost/mattermost-redux/blob/master/src/actions/users.js#L244) in `loadMe` of `mattermost-redux` is called only when `config.EnableCustomTermsOfService === 'true'`.

But since the config is not loaded, the condition will always evaluate to false and the user's terms-of-service status is not fetched on page refresh. This is why user needs to re-accept terms of service on every page refresh.

So, either we need config before API call for `loadMe` or we don't use config inside `loadMe`.

Reply:
Got it!. Thanks. Make sense to exclude them from `loadMe` and call those two api calls here after these promises are resolved IMO.

@lieut-data FYI: The config call does not take a long time to return around 200ms on pre-release. I am not sure if i am reading too much into this.

Reply:
@lieut-data @sudheerDev 
It makes sense to mention the above mentioned behavior was changed in [this PR](https://github.com/mattermost/mattermost-webapp/pull/1876).

[Here](https://github.com/mattermost/mattermost-webapp/pull/1876/files#diff-42c09aba1e6f0b3ab58dce038e383d6cL36), `getClientConfig` was called before `loadMe` and `getLicenseConfig`. 

And [here](https://github.com/mattermost/mattermost-webapp/pull/1876/files#diff-b756d78f088056823811c9d55e18baefR10), `getClientConfig`, `loadMe` and `getLicenseConfig` are called parallelly.

I couldn't figure out whether this  intentional or changed unintentionally while refactoring.

Reply:
I suspect the parallelization was done intentionally. I really like @sudheerDev's idea of conditionally doing the TOS fetch here instead of necessarily within `loadMe`, but I wouldn't be surprised if that opens up other issues, so it might be easier to defer.

We're investigating ways to fix this login/startup pipeline, but for now I'd propose that we at least continue to parallelize `getClientConfig` and `getLicenseConfig`, await on `Promise.all`, and then await on `UserActions.loadMe()`, finally returning all the promises together. That should at least preserve the parallelization of loading config and license.

Reply:
1. We shouldn't fetch TOS details here in the web app compared to redux as it would also be needed for the mobile app.
1. Updated to parallelize `getClientConfig` and `getLicenseConfig`.